src/main/java/com/gooddata/warehouse/WarehouseS3CredentialsList.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2007-2017, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata.warehouse;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.gooddata.collections.PageableList;
+import org.springframework.web.util.UriTemplate;
+
+import java.util.List;
+import java.util.Map;
+
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonTypeName("s3CredentialsList")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonDeserialize(using = WarehouseS3CredentialsListDeserializer.class)
+@JsonSerialize(using = WarehouseS3CredentialsListSerializer.class)
+public class WarehouseS3CredentialsList extends PageableList<WarehouseS3Credentials> {
```

Comment:
If we make this class package-private, then we at least need to move `URI` somewhere else because we use it for example to add it to the list of links for `Warehouse`.
I think this is also the reason why `WarehouseUsers` is public.